src/amcrest/special.py

```diff
@@ -153,3 +155,62 @@ def mjpg_stream(self, channelno=None, typeno=None, path_file=None):
                 raise CommError(error)
 
         return ret.raw
+
+    def get_event_stream(self, eventcode='VideoMotion', callback=None,
+                         timeout=(TIMEOUT_HTTP_PROTOCOL, 3600*24*365)):
+        """
+        Params:
+        eventcode:
+        string list of events
+        eg 'VideoMotion,VideoLoss'
+        NOTE: No spaces allowed!
+
+        VideoMotion: motion detection event
+        VideoLoss: video loss detection event
+        VideoBlind: video blind detection event
+        AlarmLocal: alarm detection event
+        StorageNotExist: storage not exist event
+        StorageFailure: storage failure event
+        StorageLowSpace: storage low space event
+        AlarmOutput: alarm output event
+
+        callback:
+        function to use to process data when running in a Thread
+        event data is passed as the first argument
+
+        timeout:
+        int or tuple, requests timeout format
+        no-data timeout, set to a long value for streaming
+        """
+        try:
+            ret = self.command(
+                'eventManager.cgi?action=attach&codes=[{0}]'.format(eventcode),
+                stream=True,
+                timeout_cmd=timeout
+            )
+            if ret.encoding is None:
+                ret.encoding = 'utf-8'  #required by requests when streaming
+            while True:
+                for line in ret.iter_lines(3, decode_unicode=True):
+                    if line.lower().startswith('content-length'):
+                        cont_len = int(''.join(x for x in line
+                                               if x.isdigit()))
+                        # pause the generator
+                        break
+
+                # Continue the generator and read the exact amount of the body
+                # (plus 2 as there are 2 returns included).
+                for text in ret.iter_content(cont_len+2, decode_unicode=True):
+                    if callback:
+                        callback(text.strip())
+                    else:
+                        return text.strip()
+                    break
+
+        except ConnectionError:
+            _LOGGER.debug("%s Get event stream timeout", self)
+            return None
+        except HTTPError as error:
+            _LOGGER.debug("%s Get event stream failed due "
+                          "to error: %s", self, repr(error))
```

Comment:
`self.command()` catches `requests` exceptions and throws `amcrest` specific exceptions instead. And to be consistent with the rest of the package, they should not be caught here. They should go to the caller.

Exceptions that happen during `ret.iter_content()` should be derived from `requests.RequestException`. Those should probably be caught and changed to the `amcrest` package's `CommError`. See the `snapshot()` method as an example (although that operates on the raw urllib3 stream, so HTTPError is appropriate there, whereas RequestException would be appropriate here.)